lib/aurora_bootstrapper/exporter.rb

```diff
@@ -4,30 +4,43 @@ module AuroraBootstrapper
   class Exporter
     attr_reader :client
 
-    def initialize( client:, prefix: "", export_bucket:, blacklisted_tables: "", whitelisted_tables: "", blacklisted_fields: "" )
+    def initialize( client:, prefix: "", export_bucket:, blacklisted_tables: "", whitelisted_tables: "", blacklisted_fields: "", notifier: nil )
       @match              = "#{prefix}.*"
       @export_bucket      = export_bucket
       @blacklisted_tables = blacklisted_tables.split(",")
       @whitelisted_tables = whitelisted_tables.split(",")
       @blacklisted_fields = blacklisted_fields.split(",")
       @client             = client
+      @notifier           = notifier
+      d                   = DateTime.now
+      d_str               = d.strftime("%Y-%m-%d")
+      @export_date        = ENV.fetch( 'EXPORT_DATE', d_str )
     end
 
     def export!
       @client.query( "set sql_mode='NO_BACKSLASH_ESCAPES'" )
+
       database_names.all? do | database_name |
         begin
           @client.query( "use `#{database_name}`" )
           database = Database.new database_name: database_name,
                                          client: @client,
                              blacklisted_tables: @blacklisted_tables,
                              whitelisted_tables: @whitelisted_tables,
-                             blacklisted_fields: @blacklisted_fields
+                             blacklisted_fields: @blacklisted_fields,
+                             export_date: @export_date 
           database.export! into_bucket: @export_bucket
         rescue => e
           AuroraBootstrapper.logger.error message: "Error in database #{database_name}", error: e
         end
       end
+
+      result = true
+      if @notifier
```

Comment:
Do we expect the notifier to ever not be present?

Ruby has implicit returns, so you can do
```ruby
notifier.push_state
```

Reply:
Always do null check because the world is scary :) . Sure I can change that.

Reply:
A second thought: I want to return true if notifier is null; I guess the default behavior of Ruby is return false if notifier is null?

Reply:
Can we leverage the [null object pattern](https://en.wikipedia.org/wiki/Null_object_pattern) here for the null case?

Reply:
Hmm double check the current code logic; looks like @notifier will never be null as it is initialized within the constructor. So I remove the null check logic. But it is good to know this null object pattern.